### PR TITLE
[NT-2008] Allow creator/collaborator to post

### PR DESCRIPTION
**Pull request by tobitech**
# 📲 What

Change `isBackingProject` parameter used to configure CommentComposer to `canPostComment` whose value depends on whether currently logged in user is a backer, a creator or a collaborator.

# 🤔 Why

Both creators and collaborators should be allowed to post a comment on their projects – following the same behaviour as for backers. Currently they see the message saying they need to be backers even on the project they created or are collaborating on.

# 👀 See

| Creator viewing comments on a project he created | Creator viewing comments on a project he didn't create |
| --- | --- |
|  |   |
| Collaborator viewing comments on a project he's collaborating on | Collaborator viewing comments on a project he isn't collaborating on |
|  |  |

# ✅ Acceptance criteria

1. Launch the app and ensure you're logged in with an account that has created a project or is collaborating on one.
2. Switch to dashboard tab
3. Select a project on the dashboard to view it details.
4. Tap comments to view the comments screen. As a creator or collaborator you should be able to post  a comment.
5. Navigate back to the discovery page and select another project that you neither created nor are collaborating on. If you're no backing the project you shouldn't be able to post a comment to it.

### CODE REVIEW

**Review by msadoon — APPROVED** [review]
Looks good! I couldn't check the collaborator status as discussed requires a third party to add an admin as a collaborator to a staging project.

File: Library/ViewModels/CommentsViewModel.swift [hunk 1]
```diff
@@ -87,13 +87,15 @@ public final class CommentsViewModel: CommentsViewModelType,
       .takeWhen(self.viewDidLoadProperty.signal)
       .map { project, currentUser in
         let isBacker = userIsBackingProject(project)
+        let isCreatorOrCollaborator = !project.memberData.permissions.isEmpty
```
Comment (justinswart) [inline Library/ViewModels/CommentsViewModel.swift:90]:
```swift
let isCreatorOrCollaborator = !project.memberData.permissions.isEmpty && !isBacker
```

**Review by justinswart — COMMENTED** [review]
I think we still need to be sure that they're not a backer in order for them to be a collaborator?

File: Library/ViewModels/CommentsViewModel.swift (hunk 1 above)
Comment (tobitech) [reply Library/ViewModels/CommentsViewModel.swift]: I've added this.

**Review by justinswart — APPROVED** [review]

**Pull request merged**